intake_bluesky/jsonl.py

```diff
@@ -4,15 +4,50 @@
 import pathlib
 
 from .in_memory import BlueskyInMemoryCatalog
+from .core import lastlines
 
 
 def gen(filename):
+    """
+    A JSONL file generator.
+
+    Parameters
+    ----------
+    filename: str
+        JSONL file to load.
+    """
     with open(filename, 'r') as file:
         for line in file:
             name, doc = json.loads(line)
             yield (name, doc)
 
 
+def get_stop(filename):
+    """
+    Returns the stop_doc of a Bluesky JSONL file.
+
+    The stop_doc is always the last line of the file.
+
+    Parameters
+    ----------
+    filename: str
+        JSONL file to load.
+    stop_doc: dict
+        A Bluesky run stop document.
+    """
+    stop_doc = None
+    lastline = list(lastlines(filename))[0]
+    if lastline:
+        try:
+            name, doc = json.loads(lastline)
+        except json.JSONDecodeError:
+            ...
+            # stop_doc will stay None if it can't be decoded correctly.
+        if (name == 'stop'):
```

Comment:
In the event of a decoding error, we'll end up here and `name`, never defined, will raise `NameError`. Move this block insidean `else` block, as in try/except/else, should work.